/*
 * Copyright 2009 Sun Microsystems, Inc.  All rights reserved.
 *
 * Permission is hereby granted, free of charge, to any person obtaining a
 * copy of this software and associated documentation files (the
 * "Software"), to deal in the Software without restriction, including
 * without limitation the rights to use, copy, modify, merge, publish,
 * distribute, and/or sell copies of the Software, and to permit persons
 * to whom the Software is furnished to do so, provided that the above
 * copyright notice(s) and this permission notice appear in all copies of
 * the Software and that both the above copyright notice(s) and this
 * permission notice appear in supporting documentation.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
 * OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
 * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT
 * OF THIRD PARTY RIGHTS. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR
 * HOLDERS INCLUDED IN THIS NOTICE BE LIABLE FOR ANY CLAIM, OR ANY SPECIAL
 * INDIRECT OR CONSEQUENTIAL DAMAGES, OR ANY DAMAGES WHATSOEVER RESULTING
 * FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN ACTION OF CONTRACT,
 * NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF OR IN CONNECTION
 * WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
 *
 * Except as contained in this notice, the name of a copyright holder
 * shall not be used in advertising or otherwise to promote the sale, use
 * or other dealings in this Software without prior written authorization
 * of the copyright holder.
 *
 */

Replace words that may disturb easily-offended users or easily-worried lawyers
with those that may entertain the easily-amused Sun engineer who had to edit
the list.

diff -urp -x '*~' hacks/barcode.c hacks/barcode.c
--- hacks/barcode.c	2006-04-05 23:40:32.000000000 -0700
+++ hacks/barcode.c	2009-02-05 18:58:22.542671000 -0800
@@ -102,44 +102,65 @@ static const char *words[] = 
   "addiction",
   "alertness",
   "Algeria",
+  "Amber Road",
   "anxiety",
   "aorta",
   "argyle socks",
   "attrition",
+  "automagic",
   "axis of evil",
   "bamboo",
+  "banana slug",
   "bangle",
   "bankruptcy",
   "baptism",
+  "barrelmaker",
   "beer",
   "bellicosity",
   "bells",
   "belly",
+  "Beowawe",
+  "Berkeley",
+  "BFU",
+  "biodiesel",
+  "blastwave",
   "bliss",
   "bogosity",
-  "boobies",
-  "boobs",
+  "bolthole",
+  "Boole House",
   "booty",
   "bread",
+  "brickify",
+  "browncoat",
   "bubba",
   "burrito",
+  "caiman",
   "California",
   "capybara",
   "cardinality",
   "caribou",
   "carnage",
+  "Chicago",
   "children",
+  "chime",
   "chocolate",
+  "chupacabra",
+  "cinnabar",
+  "Clear View",
   "CLONE",
-  "cock",
+  "Colorado",
   "constriction",
   "contrition",
   "cop",
+  "corona",
   "corpse",
   "cowboy",
   "crabapple",
   "craziness",
+  "crossbow",
   "cthulhu",
+  "cuddle",
+  "cuddletech",
   "Death",
   "decepticon",
   "deception",
@@ -148,20 +169,28 @@ static const char *words[] = 
   "decoy",
   "defenestration",
   "democracy",
+  "dendrite",
   "dependency",
   "despair",
   "desperation",
   "disease",
   "disease",
   "doberman",
+  "dot com",
   "DOOM",
-  "dreams",
+  "DREAM",
   "dreams",
   "drugs",
+  "Duckwater",
+  "duraflame",
   "easy",
   "ebony",
+  "ecoresponsibility",
   "election",
   "eloquence",
+  "Elko",
+  "Ely",
+  "embiggen",
   "emergency",
   "eureka",
   "excommunication",
@@ -171,18 +200,27 @@ static const char *words[] = 
   "fear",
   "fever",
   "filth",
+  "fishworks",
+  "flask",
   "flatulence",
   "fluff",
   "fnord",
+  "FOX",
+  "frak",
   "freedom",
   "fruit",
-  "fruit",
+  "fugu",
   "futility",
+  "galaxy",
+  "genunix",
   "gerbils",
-  "GOD",
+  "Gerlach",
   "goggles",
   "goobers",
   "gorilla",
+  "grok",
+  "grub",
+  "Groom Lake",
   "halibut",
   "handmaid",
   "happiness",
@@ -193,23 +231,30 @@ static const char *words[] = 
   "heroin",
   "heroine",
   "hope",
+  "honeycomb",
   "hysteria",
+  "Ichthyosaur",
   "icepick",
   "identity",
   "ignorance",
   "importance",
+  "Indiana",
   "individuality",
   "inkling",
   "insurrection",
   "intoxicant",
   "ire",
   "irritant",
+  "iwashi",
   "jade",
   "jaundice",
+  "Jet",
   "Joyce",
+  "Jupiter",
   "kidney stone",
   "kitchenette",
   "kiwi",
+  "Lahontan",
   "lathe",
   "lattice",
   "lawyer",
@@ -218,22 +263,35 @@ static const char *words[] = 
   "lobbyist",
   "love",
   "lozenge",
+  "Lund",
+  "Mad Hatter",
   "magazine",
   "magnesium",
   "malfunction",
+  "Marrakesh",
+  "Mars",
   "marmot",
   "marshmallow",
+  "McGill",
+  "Menlo Park",
+  "mercurial",
   "merit",
   "merkin",
   "mescaline",
   "milk",
   "mischief",
   "mistrust",
+  "Moapa",
   "money",
   "monkey",
   "monkeybutter",
+  "Muskoka",
+  "mustang",
   "nationalism",
   "nature",
+  "Nevada",
+  "newt",
+  "Niagara",
   "neuron",
   "noise",
   "nomenclature",
@@ -242,8 +300,11 @@ static const char *words[] = 
   "ocelot",
   "offspring",
   "overseer",
+  "Pahrump",
   "pain",
   "pajamas",
+  "panic",
+  "paravon",
   "passenger",
   "passion",
   "Passover",
@@ -253,6 +314,8 @@ static const char *words[] = 
   "petticoat",
   "pharmacist",
   "PhD",
+  "pinenut",
+  "Pioche",
   "pitchfork",
   "plague",
   "Poindexter",
@@ -261,11 +324,10 @@ static const char *words[] = 
   "presidency",
   "prison",
   "prophecy",
-  "Prozac",
   "punishment",
   "punk rock",
   "punk",
-  "pussy",
+  "quahog",
   "quagmire",
   "quarantine",
   "quartz",
@@ -274,13 +336,15 @@ static const char *words[] = 
   "rage",
   "readout",
   "reality",
-  "rectum",
   "reject",
   "rejection",
+  "Reno",
   "respect",
   "revolution",
   "roadrunner",
   "rule",
+  "sagebrush",
+  "Santa Cruz",
   "savor",
   "scab",
   "scalar",
@@ -289,16 +353,21 @@ static const char *words[] = 
   "security",
   "sediment",
   "self worth",
+  "shiny",
   "sickness",
+  "sierra",
   "silicone",
+  "Sirius",
   "slack",
   "slander",
   "slavery",
   "sledgehammer",
-  "smegma",
   "smelly socks",
+  "soda",
+  "songbird",
   "sorrow",
   "space program",
+  "Sparks",
   "stamen",
   "standardization",
   "stench",
@@ -308,24 +377,34 @@ static const char *words[] = 
   "surrender",
   "surveillance",
   "synthesis",
+  "tamarack",
   "television",
   "tenant",
   "tendril",
   "terror",
   "terrorism",
   "terrorist",
+  "Tesla",
   "the impossible",
+  "the possimpible",
   "the unknown",
+  "thumper",
+  "tiger",
+  "tonic",
   "toast",
+  "Tonopah",
   "topography",
   "truism",
+  "Tsinghua",
   "turgid",
+  "uisce",
   "underbrush",
   "underling",
   "unguent",
   "unusual",
   "uplink",
   "urge",
+  "Utah",
   "valor",
   "variance",
   "vaudeville",
@@ -333,8 +412,7 @@ static const char *words[] = 
   "vegetarian",
   "venom",
   "verifiability",
-  "viagra",
-  "vibrator",
+  "vermillion",
   "victim",
   "vignette",
   "villainy",
@@ -344,25 +422,30 @@ static const char *words[] = 
   "warehouse",
   "waste",
   "waveform",
+  "Wendover",
   "whiffle ball",
   "whorl",
   "windmill",
+  "Winchester",
   "words",
   "worm",
   "worship",
   "worship",
-  "Xanax",
+  "Wyoming",
   "Xerxes",
   "Xhosa",
   "xylophone",
   "yellow",
+  "Yerington",
   "yesterday",
   "your nose",
   "Zanzibar",
   "zeal",
   "zebra",
   "zest",
-  "zinc"
+  "zettabyte",
+  "zinc",
+  "Zulu"
 };
 
 #define WORD_COUNT (sizeof(words) / sizeof(char *))
